src/mdio/converters/segy.py

```diff
@@ -494,3 +497,43 @@ def segy_to_mdio(  # noqa: C901
     )
 
     zarr.consolidate_metadata(store_nocache)
+
+
+def _calculate_live_mask_chunksize(grid: Grid) -> Sequence[int] | int:
+    """Calculate the optimal chunksize for the live mask.
+
+    Args:
+        grid: The grid to calculate the chunksize for.
+
+    Returns:
+        Either a single integer (-1) if no chunking is needed, or a sequence of integers
+        representing the optimal chunk size for each dimension of the grid.
+    """
+    if np.sum(grid.live_mask) < INT32_MAX:
```

Comment:
Mask can be still >2GB and have only 1 live sample. It won't catch the case. Below should work

```suggestion
    if grid.live_mask.nbytes < INT32_MAX:
```